distsql/request_builder_test.go

```diff
@@ -577,52 +516,55 @@ func (s *testSuite) TestRequestBuilder6(c *C) {
 		Streaming:      false,
 		TxnScope:       oracle.GlobalTxnScope,
 	}
-
-	c.Assert(actual, DeepEquals, expect)
+	require.Equal(t, expect, actual)
 }
 
-func (s *testSuite) TestRequestBuilder7(c *C) {
-	for _, replicaRead := range []kv.ReplicaReadType{
-		kv.ReplicaReadLeader,
-		kv.ReplicaReadFollower,
-		kv.ReplicaReadMixed,
+func TestRequestBuilder7(t *testing.T) {
+	for _, replicaRead := range []struct {
```

Comment:
@xhebox IIUC we should also add `t.Parallel` even if already parallelize subtests.

see also https://pkg.go.dev/testing#hdr-Subtests_and_Sub_benchmarks

> Run does not return until parallel subtests have completed, providing a way to clean up after a group of parallel tests:
> ```go
> func TestTeardownParallel(t *testing.T) {
>     // This Run will not return until the parallel tests finish.
>     t.Run("group", func(t *testing.T) {
>         t.Run("Test1", parallelTest1)
>         t.Run("Test2", parallelTest2)
>         t.Run("Test3", parallelTest3)
>     })
>     // <tear-down code>
> }
> ```

Reply:
I thought I had discussed that in one previous PR. Checking the source code, parallel sub tests will return early, but the parent test will not return early. This behavior is reasonable for me:

1. If there are lots of sub tests, it may make use of all cpu cores. Adding more and more jobs to the queue does not make much sense to me. And it may fail the race detector
> The race detector kills the program if it exceeds 8192 concurrent goroutines, so use care when running parallel tests with the -race flag set. 
2. If there are a few tests, it will finish very quickly, i.e. scheduling may be slower than the execution. I've experience many cases where multi-thread is as quick as serial-execution.

So I will choose either parallel the parent test or subtests without really benchmarking.

Reply:
Make sense. I agree we should continuously schedule tests on real bottleneck.